google/resource_kms_key_ring.go

```diff
@@ -0,0 +1,104 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/cloudkms/v1"
+	"log"
+)
+
+func resourceKmsKeyRing() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKmsKeyRingCreate,
+		Read:   resourceKmsKeyRingRead,
+		Delete: resourceKmsKeyRingDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func kmsResourceParentString(project, location string) string {
+	return fmt.Sprintf("projects/%s/locations/%s", project, location)
+}
+
+func kmsResourceParentKeyRingName(project, location, name string) string {
+	return fmt.Sprintf("%s/keyRings/%s", kmsResourceParentString(project, location), name)
+}
+
+func resourceKmsKeyRingCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+
+	parent := kmsResourceParentString(project, location)
+
+	keyRing, err := config.clientKms.Projects.Locations.KeyRings.
```

Comment:
sorry to point this out again, but mind fixing the style here and elsewhere? https://github.com/golang/go/wiki/CodeReviewComments#line-length
I think one line would be fine, two is also fine (we usually break after the left paren)

Reply:
Ah, I don't think I understood before but I see what you're saying now.  I'll change it to a single line.